src/pilot/dracclient_raid.patch

```diff
@@ -125,7 +125,19 @@
  
      def convert_physical_disks(self, physical_disks, raid_enable):
          """Converts a list of physical disks into or out of RAID mode.
-@@ -396,13 +428,19 @@
+@@ -383,6 +415,11 @@
+             utils.validate_integer_value(size_mb, 'size_mb', error_msgs)
+ 
+         virtual_disk_prop_names.append('Size')
++
++        # Boss controller validation
++        if raid_controller.startswith('AHCI.Slot'):
++            size_mb = 0
```

Comment:
I would really prefer to keep all of the BOSS specific logic in our downstream code, especially since we already have a method in assign_role to determine if the controller is a BOSS card.  Specifically, I think this logic should be in assign_role.

Reply:
ok. do you mean we dont need to add code here in dracclient_raid ?
I have added here because it will be difficult to set size to 0 if boss controller is detected and if I set size=0 from assign_role, execution fails. 
I will retest by setting size=0 in define_storage_operating_system_logical_disk() method.

Reply:
Yes, I would prefer not to add code here.  I would think we could detect the BOSS and set the size to 0 in define_storage_operating_system_logical_disk and that should work...

Reply:
I will try again